ast/check.go

```diff
@@ -487,6 +487,9 @@ func (rc *refChecker) checkRef(curr *TypeEnv, node *typeTreeNode, ref Ref, idx i
 			return newRefErrInvalid(ref[0].Location, ref, idx, exist, types.S, getOneOfForNode(node))
 		}
 
+	case Array, Object, *Set:
+		rc.env.tree.PutOne(value, rc.env.Get(value))
```

Comment:
Yea I think you're right, this case just shouldn't exist. I'll just let it hit default